src/mod.ts

```diff
@@ -22,12 +23,86 @@ export const run = async function (functionDir: string) {
     return potentialFunctionFiles;
   });
 
-  // The CLI expects a JSON payload to be output to stdout
-  // This is formalized in the `run` hook of the CLI/SDK Tech Spec:
-  // https://corp.quip.com/0gDvAsqoaaYE/Proposal-CLI-SDK-Interface#temp:C:fOC1991c5aec8994d0db01d26260
-  console.log(JSON.stringify(resp || {}));
+  return JSON.stringify(resp || {});
+};
+
+// Start a http server that listens on the provided port
+// This server exposes two routes
+//  GET  `/health`    Returns a 200 OK
+//  POST `/functions` Accepts event payload and invokes `run`
+const startServer = async function (port: number) {
+  let server;
+  try {
+    server = Deno.listen({ port });
+  } catch (e) {
+    throw new Error(`Unable to listen on ${port}, got ${e}`);
+  }
+  for await (const conn of server) {
+    // In order to not be blocking, we need to handle each connection individually
+    // without awaiting the function
+    serveHttp(conn);
+  }
+
+  async function serveHttp(conn: Deno.Conn) {
+    // This "upgrades" a network connection into an HTTP connection.
+    const httpConn = Deno.serveHttp(conn);
+    // Each request sent over the HTTP connection will be yielded as an async
+    // iterator from the HTTP connection.
+    for await (const requestEvent of httpConn) {
+      // The native HTTP server uses the web standard `Request` and `Response`
+      // objects.
+
+      const url = new URL(requestEvent.request.url);
+      // A health check route
+      if (requestEvent.request.method == "GET" && url.pathname == "/health") {
+        return requestEvent.respondWith(
+          new Response("OK", {
+            status: 200,
+          }),
+        );
+      } else if (
+        requestEvent.request.method == "POST" && url.pathname == "/functions"
+      ) {
+        const body = await requestEvent.request.text();
+        try {
+          // run the user code
+          const response = await run(url.pathname.substring(1), body);
```

Comment:
I believe this will always resolve to `"functions"`. With that being the case, it'd be clearer to hard-code this to `"functions"` I think (and not couple it to the url path), but, I'm not sure we want to do that, lemme bounce some of how this currently works off you and let me know if any of this makes sense or something we should consider.

The first arg in `run()` gets used like so:

```ts
functionDir = `file://${await Deno.realPath(functionDir)}`;
```

Currently this runtime expects to receive the full path to where the bundled files live on the file system (with "functions" appended). The lambda extension is handling [appending "functions"](https://slack-github.com/slack/slack-cloud-lambda-extension/blob/master/denoruntime/bootstrap.go#L138) to where the bundled files live.

Should we have the interface for this continue to expect a directory to look for the functions in, perhaps as a query string param? I do think we could have the extension stop appending "functions" and move that into this file though. 

Currently it's the `deno-slack-builder` that's [building them into that folder structure](https://slack-github.com/slack/slack-cloud-lambda-extension/blob/master/denoruntime/bootstrap.go#L138), but I think the idea is that we'd formalize around that, so having this runtime be explicit about where to look for the files makes sense to me if we want to pull that logic out of the lambda extension.

Reply:
all very good points. Yeah sense the intent of this file is to only work in a hosted environment. 
On the lambda side we only allow file system access to the `functions` directory as well.
I agree hard coding this to `"functions"` is the best thing to do here

Reply:
I should just test this myself, but curious, when you test/run this locally, how does that work since it's just going to look in `file://functions`? I suppose this works fine in the lambda as that's actually what the filesystem looks like.

Reply:
yeah when I test it locally I added these files to my dir
```
?? example_payload.json
?? functions/reverse.js
```
and it works